packages/curve-ui-kit/src/shared/ui/SliderInput.tsx

```diff
@@ -0,0 +1,197 @@
+  const isRange = Array.isArray(value)
+
+  // the value of the first input / left side of the slider range. Or the value of the input/slider if there is no range
+  const currentFirst = isRange ? value[0] : value
+  // the value of the second input / right side of the slider range. Or the value of the input/slider if there is no range
+  const currentSecond = isRange ? value[1] : value
+
+  /** Translate the stored value into whatever space the slider expects. */
+  const mapToSliderValue = sliderValueTransform?.toSlider ?? ((sliderValue: number) => sliderValue)
+  /** Transform slider values back into the original value space used by the inputs. */
+  const mapFromSliderValue = sliderValueTransform?.fromSlider ?? ((sliderValue: number) => sliderValue)
+
+  const sliderValue: number | RangeValue = isRange
+    ? ([mapToSliderValue(currentFirst), mapToSliderValue(currentSecond)] as RangeValue)
+    : mapToSliderValue(currentSecond)
+
+  const sliderMinValue = sliderValueTransform?.sliderMin ?? min
+  const sliderMaxValue = sliderValueTransform?.sliderMax ?? max
+  const sliderStepValue = sliderValueTransform?.sliderStep ?? step
+
+  const handleSliderChange: SliderProps['onChange'] = (_event, newValue) => {
+    if (Array.isArray(newValue)) {
+      const [first, second] = newValue
+      onChange([mapFromSliderValue(first), mapFromSliderValue(second)])
+      return
+    }
+
+    onChange(mapFromSliderValue(newValue))
+  }
+
+  const handleInputChange = (index: 0 | 1) => (newValue: string | undefined) => {
+    // Allow decimal points and commas
+    if (typeof newValue === 'string' && /[.,]$/.test(newValue)) {
+      return
+    }
+    const numericValue = Number(newValue)
+    if (Number.isNaN(numericValue)) {
+      return
+    }
+
+    if (isRange) {
+      const nextFirst = index === 0 ? numericValue : currentFirst
+      let nextSecond = index === 1 ? numericValue : currentSecond
+
+      if (nextFirst > nextSecond) {
+        // Left input (index === 0) should not be bigger than the right input value
+        if (index === 0) nextSecond = nextFirst
+        // For the right input it is allowed because when selecting "800" for example,
+        // the user first types "8" which can be smaller than the first input.
+        else return
+      }
+
+      onChange([nextFirst, nextSecond])
+      return
+    }
+
+    onChange(numericValue)
+  }
+
+  const renderInput = (inputValue: number | undefined, index: 0 | 1) => (
+    <NumericTextField
+      aria-label={ariaLabel}
+      size={sliderInputSizeMap[size]}
+      variant="filled"
+      value={toDecimal(inputValue)}
+      min={toDecimal(min)}
+      max={toDecimal(max)}
+      onChange={handleInputChange(index)}
+      disabled={disabled}
+      data-testid={inputProps?.['data-testid'] ? `${inputProps['data-testid']}-${index}` : undefined}
+      sx={{ maxWidth: sliderInputMaxWidthMap[layoutDirection] }}
+      {...inputProps}
+    />
+  )
+
+  const renderSlider = (
+    <Slider
```

Comment:
Would it make sense if we inline the (custom) `Slider` component into here, and then rename this component from `SliderInput` to `Slider`? That way we won't have two separate slider components which may cause confusion, and we're forcing everybody to use this more advanced slider component. Unless there's a specific reason somebody wants to use the other slider component and this somehow is an ill fit?